library/std/src/env.rs

```diff
@@ -327,6 +327,7 @@ impl Error for VarError {
 /// assert_eq!(env::var(key), Ok("VALUE".to_string()));
 /// ```
 #[stable(feature = "env", since = "1.0.0")]
+#[rustc_deprecated(since = "TBD", reason = "method is unsound")]
```

Comment:
Can you link to the discussion about why it is unsound and give a quick explanation why it is unsound?

Reply:
Which discussion is preferred? I'm pretty sure at this point we could assemble a book with the various discussions.

Reply:
Good point. Maybe open a tracking issue to be refenced here and write a recap there together with all discussion you can find?

Reply:
#90308?

Reply:
Do you want to defer changing the documentation of this function to another PR, or should we do it here? If we're going to do it as part of this PR then I'm happy to suggest changes to the docs.